src/gui/qgsfieldvalidator.cpp

```diff
@@ -56,7 +56,7 @@ QgsFieldValidator::QgsFieldValidator( QObject *parent, const QgsField &field, co
     {
       if ( mField.length() > 0 && mField.precision() > 0 )
       {
-        QString re = QStringLiteral( "-?\\d{0,%1}(\\.\\d{0,%2})?" ).arg( mField.length() - mField.precision() ).arg( mField.precision() );
+        QString re = QStringLiteral( "-?\\d{0,%1}([\\.,]\\d{0,%2})?" ).arg( mField.length() - mField.precision() ).arg( mField.precision() );
```

Comment:
I think this should be based on locale setting again - or someone entering "1,000" will get a value of "1.0", when we should be rejecting the string.

Reply:
yes, I though about that, the use case is a user entering a floating point, I doubt that anyone would enter a floating point number using the thousand grouping but I might be wrong.
On the other hand there is a clear advantage in accepting both comma and dots for locales that use a comma because many applications accept both and internally convert to the correct format.
I think we could probably put it in a branch for the cases when the decimalPoint is a comma as you suggested to do in the previous comment.

Reply:
What about the case where floats with thousand separator are entered through copy-paste?

Wouldn't the following code respect both, always allow `.` and only allow `,` when it's an option?
```
QString re = QStringLiteral( "-?\\d{0,%1}([\\.%2]\\d{0,%3})?" ).arg( mField.length() - mField.precision(), decimalSeparator, mField.precision() );
```

Reply:
:+1: I was going to address this, your proposal sound perfect to me.